CesiumGeospatial/include/CesiumGeospatial/GlobeFlightPath.h

```diff
@@ -0,0 +1,100 @@
+   * destination specified in Earth-Centered, Earth-Fixed coordinates.
+   *
+   * @param sourceEcef The position that the path will begin at in ECEF
+   * coordinates.
+   * @param destinationEcef The position that the path will end at in ECEF
+   * coordinates.
+   *
+   * @returns An optional type containing a \ref GlobeFlightPath object
+   * representing the generated path, if possible. If it wasn't possible to
+   * scale the input coordinates to geodetic surface coordinates on a WGS84
+   * ellipsoid, this will return \ref std::nullopt instead.
+   */
+  static std::optional<GlobeFlightPath> fromEarthCenteredEarthFixedCoordinates(
+      const glm::dvec3 sourceEcef,
+      const glm::dvec3 destinationEcef);
+
+  /**
+   * @brief Creates a new instance of \ref GlobeFlightPath from a source and
+   * destination specified in cartographic coordinates (Longitude, Latitude, and
+   * Height).
+   *
+   * @param sourceLlh The position that the path will begin at in Longitude,
+   * Latitude, and Height.
+   * @param destinationLlh The position that the path will end at in Longitude,
+   * Latitude, and Height.
+   *
+   * @returns An optional type containing a \ref GlobeFlightPath object
+   * representing the generated path, if possible. If it wasn't possible to
+   * scale the input coordinates to geodetic surface coordinates on a WGS84
+   * ellipsoid, this will return \ref std::nullopt instead.
+   */
+  static std::optional<GlobeFlightPath> fromLongitudeLatitudeHeight(
+      const Cartographic sourceLlh,
+      const Cartographic destinationLlh);
+
+  /**
+   * @brief Samples the flight path at the given percentage of its length.
+   *
+   * @param percentage The percentage of the flight path's length to sample at,
+   * where 0 is the beginning and 1 is the end. This value will be clamped to
+   * the range [0..1].
+   * @param additionalHeight The height above the earth at this position will be
+   * calculated by interpolating between the height at the beginning and end of
+   * the curve based on the value of \p percentage. This parameter specifies an
+   * additional offset to add to the height.
+   *
+   * @returns The position of the given point on this path in Earth-Centered
+   * Earth-Fixed coordinates.
+   */
+  glm::dvec3
+  getPosition(double percentage, double additionalHeight = 0.0) const;
+
+  /**
+   * @brief Returns the length in meters of this path, measured "as the crow
+   * flies" in a straight line.
+   *
+   * @remarks This is the equivalent of the magnitude of the vector
+   * (destinationEcef - sourceEcef).
+   */
+  double getLength() const;
+
+private:
+  GlobeFlightPath(
+      glm::dvec3 scaledSourceEcef,
+      glm::dvec3 scaledDestinationEcef,
+      glm::dvec3 originalSourceEcef,
+      glm::dvec3 originalDestinationEcef);
+
+  double _totalAngle;
+  double _sourceHeight;
+  double _destinationHeight;
+  double _length;
+
+  glm::dvec3 _sourceDirection;
```

Comment:
The private members should be const.

Reply:
Not sure if relevant to this class, but the C++ Core Guidelines advises against this for copyable / movable types: [C.12](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)